jdaviz/app.py

```diff
@@ -1849,6 +1852,96 @@ def _get_viewer_item(self, ref_or_id):
+        '''
+        from astropy.wcs.utils import pixel_to_pixel
+
+        if isinstance(old_parent, str):
+            old_parent = self.data_collection(old_parent)
+
+        if isinstance(new_parent, str):
+            new_parent = self.data_collection(new_parent)
+        elif new_parent is None:
+            for data in self.data_collection:
+                if data is not old_parent:
+                    new_parent = data
+                    break
+
+        # Set subset attributes to match a remaining data collection member.
+        for subset_group in self.data_collection.subset_groups:
+            # Flag to skip this
+            if subset_group.subset_state.attributes[0].parent is old_parent:
+                for att in ("att", "xatt", "yatt", "x_att", "y_att"):
+                    if hasattr(subset_group.subset_state, att):
+                        subset_att = getattr(subset_group.subset_state, att)
+                        data_components = new_parent.components
+                        if subset_att not in data_components:
+                            cid = [c for c in data_components if c.label == subset_att.label][0]
+                            setattr(subset_group.subset_state, att, cid)
+
+                # Translate bounds through WCS if needed
+                if (self.config == "imviz" and
+                        self._jdaviz_helper.plugins["Links Control"].link_type == "WCS"):
+                    # Get the correct link to use for translation
+                    roi = subset_group.subset_state.roi
+                    if type(roi) in (CircularROI, CircularAnnulusROI,
+                                     EllipticalROI, TrueCircularROI):
+                        old_xc = roi.xc
+                        old_yc = roi.yc
+                        # Convert center
+                        x, y = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                              roi.xc, roi.yc)
+                        # Can't use set_center here because CircularAnnulusROI doesn't have it
+                        roi.xc = x
+                        roi.yc = y
+
+                        for att in ("radius", "inner_radius", "outer_radius",
+                                    "radius_x", "radius_y"):
+                            # Hacky way to get new radii with point on edge of circle
+                            # Do we need to worry about using x for the radius conversion for
+                            # radius_y if there is distortion?
+                            r = getattr(roi, att, None)
+                            if r is not None:
+                                dummy_x = old_xc + r
+                                x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                        dummy_x, old_yc)
+                                new_radius = np.abs(x2 - x)
+                                setattr(roi, att, new_radius)
```

Comment:
We can't just move the center of every subset because the resolution of the new parent data might be different, so we'll need to scale the whole thing up/down. Thus, we need to change the radii as well (or min/max in x/y for rectangles).

Reply:
Instead of moving it `att` by `att`, are you able to build on Clare's work in #2496 to use sky region as intermediate object for Imviz?

While spatial subset is also used in Cubeviz, we do not allow people from removing or changing the original cube, right? So we don't have to worry about it for now?

Reply:
Right, don't have to worry about Cubeviz for now, just Specviz.